sklearn/model_selection/_validation.py

```diff
@@ -573,13 +573,31 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
             raise ValueError("error_score must be the string 'raise' or a"
                              " numeric value. (Hint: if using 'raise', please"
                              " make sure that it has been spelled correctly.)")
-
     else:
-        fit_time = time.time() - start_time
-        test_scores = _score(estimator, X_test, y_test, scorer)
-        score_time = time.time() - start_time - fit_time
-        if return_train_score:
-            train_scores = _score(estimator, X_train, y_train, scorer)
+        try:
+            fit_time = time.time() - start_time
+            test_scores = _score(estimator, X_test, y_test, scorer)
+            score_time = time.time() - start_time - fit_time
+            if return_train_score:
+                train_scores = _score(estimator, X_train, y_train, scorer)
+        except Exception:
+            if error_score == 'raise':
+                raise
+            elif isinstance(error_score, numbers.Number):
+                score_time = time.time() - start_time - fit_time
+                test_scores = {name: error_score for name in scorer}
+                if return_train_score:
+                    train_scores = test_scores.copy()
```

Comment:
should this logic move to `_score` and add an extra parameter to `_score` to handle different `error_score` cases? Also, when only one scorer fails, only that one should be 0, not others, which is not the case with this implementation I think, right?

Reply:
Shouldn't this  be still there to handle _MultimetricScorer() failure? Please check the latest patch and let me know

Reply:
We can have the whole logic handled in `_score`:

```py
try:
    if y_test is None:
        scores = scorer(estimator, X_test)
    else:
        scores = scorer(estimator, X_test, y_test)
except:
	if error_score == 'raise':
		...
```